yellowbrick/classifier/base.py

```diff
@@ -33,119 +37,159 @@
         super(ClassificationScoreVisualizer, self).__init__(
-            model, ax=ax, fig=fig, **kwargs
+            model, ax=ax, fig=fig, is_fitted=is_fitted, **kwargs
         )
 
+        self.set_params(classes=classes, encoder=encoder, force_model=force_model)
+
     @property
-    def classes_(self):
+    def colors(self):
         """
-        Proxy property to smartly access the classes from the estimator or
-        stored locally on the score visualizer for visualization.
+        Returns ``_colors`` if it exists, otherwise computes a categorical color
+        per class based on the matplotlib color cycle. If the visualizer is not
+        fitted, raises a NotFitted exception.
+
+        If subclasses require users to choose colors or have specialized color
+        handling, they should set ``_colors`` on init or during fit.
+
+        Notes
+        -----
+        Because this is a property, this docstring is for developers only.
         """
-        if self.__classes is None:
-            try:
-                return self.estimator.classes_
-            except AttributeError:
-                return None  # TODO: raise NotFittedError instead of returning None
-        return self.__classes
+        if not hasattr(self, "_colors"):
+            if not hasattr(self, "classes_"):
+                raise NotFitted("cannot determine colors before fit")
 
-    @classes_.setter
-    def classes_(self, value):
-        self.__classes = value
+            # TODO: replace with resolve_colors
+            self._colors = color_palette(None, len(self.classes_))
+        return self._colors
 
     def fit(self, X, y=None, **kwargs):
         """
+        Fit the visualizer to the specified data.
+
         Parameters
         ----------
-
         X : ndarray or DataFrame of shape n x m
             A matrix of n instances with m features
 
         y : ndarray or Series of length n
             An array or series of target or class values
 
-        kwargs: dict
-            Keyword arguments passed to Scikit-Learn API.
-
         Returns
         -------
         self : instance
             Returns the instance of the classification score visualizer
-
         """
+        # Super fits the wrapped estimator
         super(ClassificationScoreVisualizer, self).fit(X, y, **kwargs)
 
-        # Extract the classes from the estimator
-        if self.classes_ is None:
-            self.classes_ = self.estimator.classes_
+        # Extract the classes and the class counts from the target
+        self.classes_, self.class_counts_ = np.unique(y, return_counts=True)
+
+        # Ensure the classes are aligned with the estimator
+        # If they are not aligned, ignore class counts and issue a warning
+        if hasattr(self.estimator, "classes_"):
+            if not np.array_equal(self.classes_, self.estimator.classes_):
+                self.classes_ = self.estimator.classes_
+                self.class_counts_ = None
```

Comment:
Should we add a warning here? 

This happens with specific target types, e.g. multi-label target types and is primarily triggered from PRCurve. In this case, `np.unique` is incorrect since `y` is a 2D one-hot encoded array and we just can't get counts without jumping through hoops.

Reply:
I think a warning is a good idea here &mdash; alternatively we just warn on `PRCurve` if the user provides data with a multi-label target, but my sense is that this classes-check will most useful from a developer (e.g. new `ClassificationScoreVisualizer` development) perspective